specification/digitaltwins/resource-manager/Microsoft.DigitalTwins/preview/2020-03-01-preview/digitaltwins.json

```diff
@@ -1177,7 +1177,6 @@
       ],
       "properties": {
         "properties": {
-          "x-ms-client-flatten": true,
```

Comment:
>"x-ms-client-flatten": true, [](start = 10, length = 28)

May I ask why do you remove "x-ms-client-flatten": true. This will impact the generated SDK. What is the expected SDK data structure?

Reply:
The reason it is removed it to avoid flattening of `DigitalTwinsEndpointResourceProperties` which uses `"discriminator": "endpointType",` to model inheritance. If this flag is present generated SDK does not have a way to use sub-classes to set `properties` value (e.g. `EventGrid`, `EventHub` and `ServiceBus`). 
Also there is no scenario where properties fields are set directly, they should only be initialized  via sub-classes.